### PR TITLE
Show IMPP URL if there is one

### PR DESCRIPTION
**Pull request by singpolyma**
Allows a contact to list an alternate chat address or JabberID at which they can be reached.

### CODE REVIEW

File: app/widgets/ContactData/_contactdata_card.tpl [hunk 1]
```diff
@@ -64,7 +64,10 @@
 
                 {if="$contact->url != null"}
                     <i class="material-icons icon-text">link</i>
-                    {if="filter_var($contact->url, FILTER_VALIDATE_URL)"}
+                    {if="parse_url($contact->url, PHP_URL_SCHEME) == 'xmpp'"}
+
+                        <a href="{$contact->url}" onclick="MovimUtils.reload('{$c->route('contact', substr($contact->url, 5))}'); return false" class="u-impp" rel="me">{$contact->url}</a>
```
Comment (edhelas) [inline app/widgets/ContactData/_contactdata_card.tpl]: Why the "u-impp" class and `return: false` ?

File: app/Contact.php [hunk 2]
```diff
@@ -229,6 +229,9 @@ public function setVcard4($vcard)
         $this->url = !empty($vcard->url->uri)
             ? (string)$vcard->url->uri
             : null;
+        if (!$this->url && $vcard->impp->uri) {
```
Comment (edhelas) [inline app/Contact.php]: I tend to add a carriage return before if() for readability.
Also testing `$this->url == null` seems cleaner :)

**Comment by singpolyma** [comment]
Somebody claiming to be Jaussoin Timothée wrote:
>@edhelas commented on this pull request.
>
>
>
>> @@ -64,7 +64,10 @@
>
>                 {if="$contact->url != null"}
>                     <i class="material-icons icon-text">link</i>
>-                    {if="filter_var($contact->url, FILTER_VALIDATE_URL)"}
>+                    {if="parse_url($contact->url, PHP_URL_SCHEME) == 'xmpp'"}
>+
>+                        <a href="{$contact->url}" onclick="MovimUtils.reload('{$c->route('contact', substr($contact->url, 5))}'); return false" class="u-impp" rel="me">{$contact->url}</a>
>
>Why the "u-impp" class and `return: false` ?

u-impp is just from h-card, it can be removed if not wanted.

return false is to make it use the soft reload instead of following the link 
after
